### PR TITLE
Check that stored certificate chain matches certificate

### PR DESCRIPTION
**Pull request by intgr**
Similar to how the private key is checked in #2269.

The second and final part of what was originally #1514.

### CODE REVIEW

File: lemur/common/validators.py [hunk 1]
```diff
@@ -138,3 +125,35 @@ def verify_private_key_match(key, cert, error_class=ValidationError):
     """
     if key.public_key().public_numbers() != cert.public_key().public_numbers():
         raise error_class("Private key does not match certificate.")
+
+
+def verify_cert_chain(certs, error_class=ValidationError):
+    """
+    Verifies that the certificates in the chain are correct.
```
Comment (intgr) [inline lemur/common/validators.py]: This is exactly the same function as used in #2262. It doesn't matter which one gets merged first.

**Comment by intgr** [comment]
Ping?

File: lemur/tests/factories.py [hunk 2]
```diff
@@ -140,6 +140,7 @@ class CACertificateFactory(CertificateFactory):
 class InvalidCertificateFactory(CertificateFactory):
     body = INVALID_CERT_STR
     private_key = ''
+    chain = ''
```
Comment (intgr) [inline lemur/tests/factories.py]: Needed to override this, otherwise it would cause `AssertionError` in `Certificate.check_integrity`

**Comment by hosseinsh** [comment]
This looks good to us, there is just a conflict to resolve.

**Comment by intgr** [comment]
Cool. Rebased & resolved conflicts.

**Comment by hosseinsh** [comment]
Thanks Marti for the PR. 
Just merged it.

**Pull request merged**